tests/unit/test_task_8_9.py

```diff
@@ -0,0 +1,156 @@
+from unittest.mock import patch
+import pytest
+
+import lab8_9.subtask_1
+import lab8_9.subtask_2
+import lab8_9.subtask_3
+import lab8_9.subtask_4
+import lab8_9.subtask_5_6
+import lab8_9.subtask_7
+import lab8_9.subtask_8
+import lab8_9.subtask_9
+import lab8_9.subtask_10
+import lab8_9.subtask_11
+import lab8_9.subtask_12
+
+
+@pytest.mark.parametrize('input_param, expected',
+                         [(['Sophia', 'Madison', 'Admin'],
+                           ['Sophia, thank you for logging in again..',
+                            'Madison, thank you for logging in again..',
+                            'Admin, I hope you\'re well']),
+                          ([], ['We need to find some users!'])])
+def test_subtask_1(input_param, expected):
+    result = lab8_9.subtask_1.get_greetings(input_param)
+    assert result == expected
+
+
+@pytest.mark.parametrize('input_param, expected', [(1, None),
+                                                   (4, 'quadrilateral'),
+                                                   (3, 'triangle')])
+def test_subtask_2(input_param, expected):
+    result = lab8_9.subtask_2.get_figure_name(input_param)
+    assert result == expected
+
+
+@pytest.mark.parametrize('input_param, expected', [(list(range(1, 5)),
+                                                    ['1st', '2nd', '3rd', '4th', ])])
+def test_subtask_3(input_param, expected):
+    result = lab8_9.subtask_3.get_number_names(input_param)
+    assert result == expected
+
+
+@pytest.mark.parametrize('input_param, expected', [(4, 'The number is even'),
+                                                   (11, 'The number is odd')])
+def test_subtask_4(input_param, expected):
+    result = lab8_9.subtask_4.get_parity(input_param)
+    assert result == expected
+
+
+@pytest.mark.parametrize('input_param, expected', [(2024, True), (2003, False)])
+def test_subtask_5_6_is_leap_year(input_param, expected):
+    result = lab8_9.subtask_5_6.is_leap_year(input_param)
+    assert result == expected
+
+
+@pytest.mark.parametrize('input_param, expected', [(('mAy', 1984), 31),
+                                                   (('February', 2024), 29)])
+@patch('lab8_9.subtask_5_6.is_leap_year')
+def test_subtask_5_6_get_days_or_default_leap_years(mock, input_param, expected):
+    mock.return_value = True
+    result = lab8_9.subtask_5_6.get_days_or_default(input_param[0], input_param[1])
+    assert result == expected
+
+
+@pytest.mark.parametrize('input_param, expected', [(('augustember', 2005), None),
+                                                   (('february', 2003), 28)])
+@patch('lab8_9.subtask_5_6.is_leap_year')
+def test_subtask_5_6_get_days_or_default_not_leap_years(mock, input_param, expected):
+    mock.return_value = False
+    result = lab8_9.subtask_5_6.get_days_or_default(input_param[0], input_param[1])
+    assert result == expected
```

Comment:
How do I test different cases of mocking is_leap_year properly? Or do I just split them like this?

Reply:
you were on the right way